### ISSUE TITLE
clippy throws correctness errors: you are implementing `Hash` explicitly but have derived `PartialEq`

### ISSUE THREAD

**Issue by guidorice** [georust/geo#391]
I am seeing this on master. I think it was introduced in #389. Here is one of the errors:

```
error: you are implementing `Hash` explicitly but have derived `PartialEq`
   --> geo-types/src/line.rs:164:1
    |
164 | / impl<T: CoordinateType + Hash> Hash for Line<T> {
165 | |     fn hash<H: Hasher>(&self, state: &mut H) {
166 | |         self.start.hash(state);
167 | |         self.end.hash(state);
168 | |     }
169 | | }
    | |_^
    |
note: `PartialEq` implemented here
   --> geo-types/src/line.rs:5:10
    |
5   | #[derive(PartialEq, Clone, Copy, Debug)]
    |          ^^^^^^^^^
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#derive_hash_xor_eq
```

**Comment by guidorice** [comment]
I think both `PartialEq` and `Eq` need to be implemented everywhere Hash was implemented.

Here is someone's blog post that's related to this:

https://www.philipdaniels.com/blog/2019/rust-equality-and-ordering/

The clippy error page is

https://rust-lang.github.io/rust-clippy/master/index.html#derive_hash_xor_eq

**Comment by frewsxcv** [comment]
Thanks for reporting this! Fixed in #395
